### PR TITLE
Properly handle missing READ permission

### PR DESCRIPTION
**Pull request by MorrisJobke**
* downstream of https://github.com/owncloud/core/pull/27245

cc @rullzer @icewind1991 @LukasReschke

### CODE REVIEW

**Comment by icewind1991** [comment]
I feel that throwing forbidden would be better than pretending the file doesn't exist

File: apps/dav/lib/Connector/Sabre/Directory.php [hunk 1]
```diff
@@ -227,6 +233,11 @@ public function getChildren() {
 			return $this->dirContent;
 		}
 		try {
+			if (!$this->info->isReadable()) {
+				// return 403 instead of 404 because a 404 would make
+				// the caller believe that the collection itself does not exist
+				throw new Forbidden('No read permissions');
```
Comment (nickvergessen) [inline apps/dav/lib/Connector/Sabre/Directory.php]: above the comment is `avoid detecting files through this way` 
and now the information is leaked?

Sounds strange to me.

File: apps/dav/lib/Connector/Sabre/FilesPlugin.php [hunk 2]
```diff
@@ -287,6 +287,10 @@ public function handleGetProperties(PropFind $propFind, \Sabre\DAV\INode $node)
 		$httpRequest = $this->server->httpRequest;
 
 		if ($node instanceof \OCA\DAV\Connector\Sabre\Node) {
+			if (!$node->getFileInfo()->isReadable()) {
+				// avoid detecting files through this means
+				throw new NotFound();
```
Comment (nickvergessen) [inline apps/dav/lib/Connector/Sabre/FilesPlugin.php]: missing from docs

File: apps/dav/lib/Connector/Sabre/File.php [hunk 3]
```diff
@@ -307,6 +308,10 @@ private function emitPostHooks($exists, $path = null) {
 	public function get() {
 		//throw exception if encryption is disabled but files are still encrypted
 		try {
+			if (!$this->info->isReadable()) {
+				// do a if the file did not exist
+				throw new NotFound();
```
Comment (nickvergessen) [inline apps/dav/lib/Connector/Sabre/File.php]: missing from docs

File: apps/dav/lib/Connector/Sabre/Directory.php (hunk 1 above)
Comment (nickvergessen) [inline apps/dav/lib/Connector/Sabre/Directory.php]: missing from docs

**Review by nickvergessen — CHANGES_REQUESTED** [review]
With this change, I can not navigate into folders anymore, that are shared as "drop folders"

**Pull request closed without merging**